File: 2486-MostFrequentEvenElement/2486-MostFrequentEvenElement.js
// Last updated: 3/7/2026, 10:02:17 PM
/**
 * @param {number[]} nums
 * @return {number}
 */
var mostFrequentEven = function (nums) {
    let obj = {};
    let count = 0;
    let res = -1;

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] % 2 == 0) {
            if (obj[nums[i]] === undefined) {
                obj[nums[i]] = 1;
            }
            else {
                obj[nums[i]]++;
            }
        }
    }
    console.log(obj)

    for(let key in obj){
        if(obj[key] > count){
            count = obj[key];
            res = Number(key);
        }
    }
    return res;
};

File: 2486-MostFrequentEvenElement/2486-MostFrequentEvenElement.cpp
// Last updated: 8/18/2026, 11:47:26 AM
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        unordered_map<int, int> freq;

        for(int i=0; i<nums.size(); i++){
            freq[nums[i]]++;
        }

        int count = 0;
        int res = -1;

        for(auto it : freq){
            if(it.first % 2 == 0){
                if(it.second > count){
                    count = it.second;
                    res = it.first;
                }
                else if(it.second == count && it.first < res){
                    res = it.first;
                }
            }
        }
        return res;
    }
};